components/collector/src/source_collectors/gitlab/base.py

```diff
@@ -201,26 +205,69 @@ async def _landing_url(self, responses: SourceResponses) -> URL:
         pipelines = await self._pipelines(responses)
         return URL(pipelines[0].web_url)
 
+    async def _get_source_responses(self, *urls: URL) -> SourceResponses:
+        """Extend to get pipeline schedule descriptions."""
+        # Get the pipeline schedule descriptions so the user can filter scheduled pipelines by schedule description
+        # The pipeline schedule descriptions is a mapping from pipeline ids to schedule descriptions
+        self.pipeline_schedule_descriptions = await self._scheduled_pipeline_descriptions()
+        return await super()._get_source_responses(*urls)
+
+    async def _parse_entities(self, responses: SourceResponses) -> Entities:
+        """Parse the entities from the responses."""
+        return Entities(
+            [
+                Entity(
+                    key=str(pipeline.id),
+                    name=pipeline.name,
+                    ref=pipeline.ref,
+                    status=pipeline.status,
+                    trigger=pipeline.source,
+                    schedule=pipeline.schedule_description,
+                    created=pipeline.created_at,
+                    updated=pipeline.updated_at,
+                    duration=str(minutes(pipeline.duration)),
+                )
+                for pipeline in await self._pipelines(responses)
+            ],
+        )
+
+    def _include_entity(self, entity: Entity) -> bool:
+        """Return whether this entity should be considered."""
+        branches = self._parameter("branches")
+        matches_branches = match_string_or_regular_expression(entity["ref"], branches) if branches else True
+        schedule_descriptions = self._parameter("pipeline_schedules_to_include")
+        matches_schedule_description = (
+            match_string_or_regular_expression(entity["schedule"], schedule_descriptions)
+            if schedule_descriptions
+            else True
+        )
+        matches_status = entity["status"] in self._parameter("pipeline_statuses_to_include")
+        matches_trigger = entity["trigger"] in self._parameter("pipeline_triggers_to_include")
+        return matches_branches and matches_schedule_description and matches_status and matches_trigger
+
     async def _pipelines(self, responses: SourceResponses) -> list[Pipeline]:
         """Get the pipelines from the responses."""
         pipelines = []
-        try:
-            for response in responses:
-                json = await response.json()
-                # The GitLab pipelines endpoint returns a list or, if a pipeline id is passed, one pipeline. Harmonize:
-                json_list = json if isinstance(json, list) else [json]
-                pipelines.extend([Pipeline(**pipeline) for pipeline in json_list])
-        except StopAsyncIteration:
```

Comment:
We no longer need to guard against the `StopAsyncIteration`?

Reply:
Apparently, the need to catch `StopAsyncIteration` was caused by our test code. It doesn't occur (so far) with testing against a live GitLab.